src/lib.rs

```diff
@@ -185,6 +185,22 @@ enum_rename!{
     }
 }
 
+enum_rename!{
```

Comment:
I think we should start moving `enum_rename` uses to the more powerful (because it can apply feature cfgs) proc macro, `llvm_enum` which does this and more. See `InstructionOpcode` for example: https://github.com/TheDan64/inkwell/blob/master/src/values/instruction_value.rs#L16-L18

Once that's done, could you add doc comments to each variant and briefly describe what each one is used for?  IE

```rust
#[llvm_enum(LLVMAtomicRMWBinOp)]
pub enum AtomicRMWBinOp {
    /// Used to add two things, atomically
    Add,
```

Reply:
Done.